File: lecture_012(Sorting)/l_003_selectionsort.java
import java.util.Scanner;
public class l_003_selectionsort {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        selectionSort(arr);
    }
    public static void display(int[] arr) {
        for(int r: arr) {
            System.out.print(r+" ");
        }
        System.out.println();
    }
    public static void input(int[] arr) {
        for(int i=0; i<arr.length; i++) {
            arr[i] = scn.nextInt();
        }
        display(arr);
    }
    public static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
    public static void selectionSort(int[] arr) {
        for(int i=0; i<arr.length-1; i++) {
            int min = i;
            for(int j=i+1; j<arr.length; j++) {
                if(arr[min]>arr[j]) {
                    min = j;
                }
            }
            swap(arr,i,min);
        }
        display(arr);
    }
}

File: lecture_012(Sorting)/l_003_selectionsort.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<int> &arr) {
    for (int i:arr) {        
        cout<<i<<" ";         
    }
    cout<<endl;
}
void input(vector<int> &arr) {   
    for(int i=0; i<arr.size(); i++) {   
        cin>>arr[i];
    }
    display(arr);
}
void swap(vector<int> &arr, int i, int j) {
    int temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}
void selectionsort(vector<int> &arr) {
    for(int i=0; i<arr.size()-1; i++) {
        int min = i;
        for(int j=i+1; j<arr.size(); j++) {
            if(arr[min]>arr[j]) {
                min = j;
            }
        }
        swap(arr,i,min);
    }
    display(arr);
}

int main() {
    int n;
    cin>>n;
    vector<int>arr(n,0);   
    input(arr); 
    selectionsort(arr);

    return 0;                   
}
